client-app/src/pages/account/store/actions.ts

```diff
@@ -1,22 +1,26 @@
-import AccountService from "@common/account.service";
+import Vue from "vue";
 import { ActionTree } from "vuex";
 import { FETCH_PROFILE, SET_PROFILE, UPDATE_USER } from "@account/store/definitions";
 import { State } from "@account/store/state";
+import { ApiAccountClient, UserUpdateInfo } from '@common/api/api.client';
 
 //actions
 export const actions: ActionTree<State, State> = {
   async [FETCH_PROFILE](context) {
     try {
       context.commit(FETCH_PROFILE);
-      const { data } = await AccountService.getUserAccount();
-      context.commit(SET_PROFILE, data);
-      return data;
+      const accountClient = new ApiAccountClient(Vue.axios.defaults.baseURL, Vue.axios);
+      const user = await accountClient.getCurrentUser(Vue.$storeInfo.storeName, Vue.$storeInfo.currentLng);
+      context.commit(SET_PROFILE, user);
+      return user;
     } catch (e) {
       // #todo SET_ERROR cannot work in multiple states
       // context.commit(SET_ERROR, response.data.errors)
     }
   },
-  [UPDATE_USER]() {
-    //todo
+  async [UPDATE_USER](context , userUpdateInfo: UserUpdateInfo) {
+    const accountClient = new ApiAccountClient(Vue.axios.defaults.baseURL, Vue.axios);
```

Comment:
I think is the better will be defined `ApiAccountClient` as external service  and inject it by import 
```JS
 import { apiAccountClient } from '@common/api.services'
```
instead of using new each time

Reply:
Shortly: what about creating singleton

Reply:
yeh, singleton definitely

Reply:
In such case, it is better to generate one client class instead of several. Is that right?

Reply:
I don't think so. We may still generate multiple client classes. Just don't create multiple instances, use singleton.
@tatarincev My proposal is to use the following dependency injection: https://github.com/inversify/InversifyJS